### PR TITLE
Implement wp-test-utils library. 

### LINKED ISSUE #6609: Adopt PHPUnit polyfills
## Task Description

Consider adopting https://github.com/Yoast/PHPUnit-Polyfills to make our lives easier writing PHP tests.

Perhaps via https://github.com/Yoast/wp-test-utils

Related: #5473

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Implement [wp-test-utils](https://github.com/Yoast/wp-test-utils) library to allow for cross compatibility with different PHPUnit versions. 

## Relevant Technical Choices

Create a new a `Test_Case` and `Test_REST_TestCase` abstract classes. These extend, `Yoast\WPTestUtils\WPIntegration\TestCase` class that intern extends, `WP_UnitTestCase`. 

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6609

### CODE REVIEW

**Comment by miina** [comment]
@spacedmonkey Looks like PHP Unit tests aren't passing on this PR? 1 seems to be failing and the others seem to be Cancelled.

File: tests/phpunit/tests/AMP/Optimization.php [hunk 1]
```diff
@@ -99,6 +98,7 @@ public function test_get_optimizer_configuration_no_ssr() {
 
 		$transformers = [
 			AmpBoilerplate::class,
+			AmpBoilerplateErrorHandler::class,
```
Comment (spacedmonkey) [inline tests/phpunit/tests/AMP/Optimization.php]:
```suggestion
```

**Comment by spacedmonkey** [comment]
@miina Thanks for flagging. It is fixed now

File: tests/phpunit/includes/Private_Access.php [hunk 2]
```diff
@@ -38,7 +38,7 @@ trait Private_Access {
 	 * @return mixed Return value of the method call.
 	 * @throws ReflectionException If the object could not be reflected upon.
 	 */
-	private function call_private_method( $object, $method_name, $args = [] ) {
+	protected function call_private_method( $object, $method_name, $args = [] ) {
```
Comment (miina) [inline tests/phpunit/includes/Private_Access.php:41]: Curious, what's the reason for a change here?

Comment (spacedmonkey) [reply tests/phpunit/includes/Private_Access.php:41]: Now this trait is being used in a Test_Case class, that is a parent class meaning this is not accessible anymore. Private method can not use child classes.

**Comment by miina** [comment]
@spacedmonkey Looks good to me! One more question first:
Should there be an update in the unit tests documentation as well? To maybe add a reference to the `wp-test-utils` and mention that this is in use for compatibility? Thoughts?

**Comment by spacedmonkey** [comment]
> Should there be an update in the unit tests documentation as well? To maybe add a reference to the wp-test-utils and mention that this is in use for compatibility? Thoughts?

@miina Good scout. I have updated the docs.

**Review by miina — APPROVED** [review]

File: composer.json [hunk 3]
```diff
@@ -34,7 +34,8 @@
     "phpunit/phpunit": "^5.7 || ^6.5 || ^7.5",
```
Comment (swissspidy) [inline composer.json:34]: @spacedmonkey My understanding was that we can now remove this line here, as the wp-test-utils library will automatically install it.

**Pull request merged**